### DISCUSSION TITLE
iPhone für Mitteilungen registrieren

### DISCUSSION THREAD

**Discussion by 0x636b** [brutella/hkknx-public#238] [Q&A] [answered]
Hallo zusammen.

Ich schaffe es leider nicht, ein zweites iPhone als Gerät zu registrieren, damit dieses auch Mitteilungen empfangen kann.
Sobald ich das bereits registrierte iPhone lösche, ist die Registrierung des anderen erfolgreich.

Die Meldung in der Configurator-App ist immer "...erfolgreich registriert", unter den Einstellungen wird jedoch nur eins gelistet.

Momentan nutze ich noch die "Test"-Version, kann es ggf. auch daran liegen?

Viele Grüße
0x636b

**Comment by brutella** [comment, marked as answer]
> Momentan nutze ich noch die "Test"-Version, kann es ggf. auch daran liegen?

~Nein, das sollte keine Probleme machen.  Haben beide iPhones den selben Namen?~

*Edit:* Das Problem ist mit Version 1.2.3 der [Konfigurator-App](https://apps.apple.com/us/app/home-configurator/id1525371546) behoben.

**Reply by 0x636b** [reply]
Die beiden iPhones besitzen nicht denselben Namen. 
Allerdings erscheint beim Registrieren jedes Mal nur "iPhone" und nicht der korrekte Name des jeweiligen iPhones. Den Namen passe ich dann an, jedoch ist auch danach das Registrieren nicht erfolgreich.

Ich habe in der Zwischenzeit die Lizenz erworben, dass es also an der Test-Version liegt, kann ich somit ausschließen.

Setup:
iPhone 12
iPhone 13
zwei untersch. iCloud-Accounts

**Reply by brutella** [reply]
Welche Namen tauchen unter **Einstellungen** → **Apple-Geräte** auf, wenn du das iPhone 12 oder das iPhone 13 registrierst?

**Reply by 0x636b** [reply]
Es taucht einfach nur "iPhone" auf.
Ich habe es eben einmal mit zwei iPads getestet. Diese werden beide als weitere Geräte problemlos registriert, das erste nur als "iPad", (iPad Pro, iPadOS 16.1) das zweite dann mit dem korrekten Namen (iPad Air 2, iPadOS 15.7.1)

Ich schaue mir das heute Abend bei dem iPhone meiner Frau noch einmal genauer an, installiere die "Configurator"-App einmal neu und versuche es dann nochmal.

**Reply by 0x636b** [reply]
Ein frohes neues Jahr. :)

Ich bin noch eine Antwort schuldig:
Nach einigem Ausprobieren schaffe ich es nicht, das iPhone meiner Frau unter den registrierten Geräten angezeigt zu bekommen.
Dennoch habe ich in der Zwischenzeit festgestellt, dass das iPhone Mitteilungen z. Bsp. aus Skripten erhält.
Also: Ziel erreicht. :)

Es ist mit keinem anderen meiner Geräte reproduzierbar, da alle sauber angezeigt werden.
Der einzige Unterschied zwischen den Geräten, der offensichtlich ist: der iCloud-Account...

**Reply by brutella** [reply]
Ich bin mir sicher, dass es an den Namen der Geräte liegt. Der Gerätenamen wird in den iOS Einstellungen → Allgemein → Info → Name angezeigt.

**Reply by 0x636b** [reply]
Mein iPhone heißt "Christophs-iPhone", das meiner Frau "Tanjas iPhone". Beim Registrieren steht nur einmal "iPhone" da. Die Nachrichten empfängt das iPhone meiner Frau nach wie vor, obwohl es eben nicht in der Liste registrierter Geräte steht. Auch wenn ich das vorhandene iPhone aus der Liste lösche, erhalte ich die Nachrichten nicht mehr, meine Frau jedoch schon. Auch ein Ändern der Namen der iPhones (ohne Sonder- oder Leerzeichen) schafft keine Abhilfe.

Auch seltsam:
Die "Configurator"-App auf dem iPhone meiner Frau zeigt nicht "Nach HKKNX suchen an", die Registrierung kann ich einzig über den Link auf der Website unter den Einstellungen durchführen. ("Geräte im lokalen Netzwerk finden" ist erlaubt.) Auch eine Neuinstallation der App ändert nichts.

**Reply by brutella** [reply]
Hab gerade entdeckt, dass seit iOS 16 für Apps alle iPhones nur noch "iPhone" heißen. Der benutzerdefinierte Name ist für Apps (vermutlich aus Gründen der Privatsphäre) ohne [spezielles Entitlement](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name) nicht mehr auslesbar.

Die Chancen sind nicht hoch, dass ich dieses Entitlement bekomme. Werde daher einen Workaround einbauen müssen.

**Reply by brutella** [reply]
Mit der neuen Version 1.2.3 der [Konfigurator-App](https://apps.apple.com/us/app/home-configurator/id1525371546) kann beim Registrieren der Name des Gerätes vergeben werden. Hier sollte ein eindeutiger Name vergeben werden. Damit sollte das Problem gelöst sein.

**Reply by 0x636b** [reply]
Vielen Dank für die schnelle Umsetzung. Es funktioniert einwandfrei! :)
